test/test-data-unicode-set.json

```diff
@@ -1579,5 +1513,119 @@
     "input": "[\\__]"
+  },
+  "[&&]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [&&]\n       ^",
+    "input": "[&&]"
+  },
+  "[!!]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [!!]\n       ^",
+    "input": "[!!]"
+  },
+  "[##]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [##]\n       ^",
+    "input": "[##]"
+  },
+  "[$$]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [$$]\n       ^",
+    "input": "[$$]"
+  },
+  "[%%]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [%%]\n       ^",
+    "input": "[%%]"
+  },
+  "[**]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [**]\n       ^",
+    "input": "[**]"
+  },
+  "[++]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [++]\n       ^",
+    "input": "[++]"
+  },
+  "[,,]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [,,]\n       ^",
+    "input": "[,,]"
+  },
+  "[..]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [..]\n       ^",
+    "input": "[..]"
+  },
+  "[::]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [::]\n       ^",
+    "input": "[::]"
+  },
+  "[;;]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [;;]\n       ^",
+    "input": "[;;]"
+  },
+  "[<<]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [<<]\n       ^",
+    "input": "[<<]"
+  },
+  "[==]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [==]\n       ^",
+    "input": "[==]"
+  },
+  "[>>]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [>>]\n       ^",
+    "input": "[>>]"
+  },
+  "[??]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [??]\n       ^",
+    "input": "[??]"
+  },
+  "[@@]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "Invalid set operation in character class at position 3\n    [@@]\n       ^",
+    "input": "[@@]"
+  },
+  "[^^^]": {
```

Comment:
`[^^]/v` is actually valid due to the negation. To test the reserved double punctuator we have to construct a case where `^^` is seen in the middle.